azurerm/resource_arm_subnet.go

```diff
@@ -416,6 +429,18 @@ func expandSubnetDelegation(d *schema.ResourceData) []network.Delegation {
 	return retDelegations
 }
 
+func expandSubnetNatGateway(d *schema.ResourceData) network.SubResource {
+	id := d.Get("nat_gateway_id").(string)
+	subResource := network.SubResource{
+		ID: utils.String(id),
+	}
+	return subResource
+}
```

Comment:
rather than adding this association within the Subnet resource, to ensure this is removed prior to the Subnet being deleted this'll instead need to be a separate 'association' resource [example](https://www.terraform.io/docs/providers/azurerm/r/subnet_route_table_association.html) - can we remove this field from the Subnet resource and instead make this an

Reply:
I reverted changes in `subnet` and created a new resource called `azurerm_subnet_nat_gateway_association` for this purpose